src/api/getAllConversations.ts

```diff
@@ -0,0 +1,29 @@
+import { idbDatabase, CONVERSATION_STORE_NAME } from './idb';
+import type { ConversationHistory } from '../types';
+
+export async function getAllConversations(): Promise<
+  Record<string, ConversationHistory>
+> {
+  const db = await idbDatabase();
+  const tx = db.transaction(CONVERSATION_STORE_NAME, 'readonly');
+  const store = tx.objectStore(CONVERSATION_STORE_NAME);
+  const request = store.getAll();
+
+  return new Promise((resolve, reject) => {
+    request.addEventListener('success', () => {
+      const results: ConversationHistory[] = request.result;
+      resolve(
+        results.reduce(
+          (acc, curr) => {
+            acc[curr.id] = curr;
+            return acc;
+          },
+          {} as Record<string, ConversationHistory>,
+        ),
+      );
```

Comment:
This would be the `Record` not an array, right?

Reply:
correct, an object of conversations with keys as the id